Options.py

```diff
@@ -134,6 +134,18 @@ def get_option_name(cls, value: T) -> str:
         else:
             return cls.name_lookup[value]
 
+    @classmethod
+    def get_choice(cls, option_weights: typing.Dict[typing.Any]) -> Option[T]:
```

Comment:
This typing doesn't make sense. We say this parameter needs to be dict, but then the function looks like it supports stuff that's not dict.
The parameter name `option_weights` also doesn't seem right. Judging by what I see in the unit test, it could sometimes be weights and sometimes not?
Is it a `yaml_option_entry` maybe?

I would also like to see a docstring explaining in more detail what the job of this function is (like explaining what the parameter could be and how it will be interpreted).

Reply:
I forgot to change the typing when I changed the args. Could probably provide proper typing, but it can be `str`, `int`, `list[any]`, `dict[any]`, so unsure how useful that would be.